### PR TITLE
feat(pie-css): DSW-000 added media query shortcuts using include-media

### PR DESCRIPTION
**Pull request by ashleynolan**
## Describe your changes (can list changeset entries if preferable)

- [Added] - Media query shortcuts using include-media

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have added thorough tests where applicable (unit / component / visual)
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If changes will affect consumers of the package, I have created a changeset entry.
- [x] If a changeset file has been created, I have used the `/snapit` functionality to test my changes in a consuming application

### Reviewer 1
- [x] I have reviewed the updated README

### Reviewer 2
- [x] I have reviewed the updated README

### CODE REVIEW

File: packages/tools/pie-css/scss/settings/_breakpoints.scss [hunk 1]
```diff
@@ -0,0 +1,15 @@
+$unit-intervals: (
+    'px': 1,
+    'em': 0.001,
+    // We need more sensitivity, some browsers are very specific with breakpoints
+    'rem': 0.1,
+    '': 0
+);
+$breakpoints: (
```
Comment (jamieomaguire) [inline packages/tools/pie-css/scss/settings/_breakpoints.scss:8]: NIT: Please separate these with new lines

File: packages/tools/pie-css/scss/settings/index.scss [hunk 2]
```diff
@@ -0,0 +1 @@
+@forward './mq';
```
Comment (jamieomaguire) [inline packages/tools/pie-css/scss/settings/index.scss:1]: I think we need a readme section on how and why to use this

File: packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+          }
+        `);
+
+        // Act
+        const css = stripCSSWhitespace(compileCss(scssToTest));
+
+        // Assert
+        expect(css).toBe(expectedCss);
+    });
+    it('should render the expected CSS media query', () => {
```
Comment (jamieomaguire) [inline packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts:37]: NIT: Please separate with new lines between tests

File: packages/tools/pie-css/scss/settings/_breakpoints.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
+$unit-intervals: (
+    'px': 1,
+    'em': 0.001,
+    // We need more sensitivity, some browsers are very specific with breakpoints
+    'rem': 0.1,
+    '': 0
+);
```
Comment (jamieomaguire) [inline packages/tools/pie-css/scss/settings/_breakpoints.scss:7]: Please could you add a comment explaining what these are?

File: packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+        // Assert
+        expect(css).toBe(expectedCss);
+    });
+    it('should render the xs media query breakpoint shortcut', () => {
+        // Arrange
+        const scssToTest = `
+          @use 'settings' as *;
+
+          .foo {
+            @include media('>=xs') {
```
Comment (jamieomaguire) [inline packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts:69]: The test names do not mention the use of `>=` - should that be included and do we want to test all variations?

File: packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+
+          .foo {
+            @media (max-width: map.get($breakpoints, "sm")) {
+                color: red;
+            }
+          }
+        `;
+
+        const expectedCss = stripCSSWhitespace(`
+          @media (max-width:600px) {
```
Comment (jamieomaguire) [inline packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts:24]:
```suggestion
          @media (max-width: 600px) {
```

File: packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+            }
+          }
+        `);
+
+        // Act
+        const css = stripCSSWhitespace(compileCss(scssToTest));
+
+        // Assert
+        expect(css).toBe(expectedCss);
+    });
```
Comment (jamieomaguire) [inline packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts:218]: There is a lot of similar code in this file. Could we do some kind of `it.each` syntax to reduce code?

File: packages/tools/pie-css/scss/settings/index.scss (hunk 2 above)
Comment (ashleynolan) [reply packages/tools/pie-css/scss/settings/index.scss:1]: I'll add some info into the pie-css `README` once I've tested it (incase it changes) – will add to this PR though when ready to merge 👍

File: packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts (hunk 7 above)
Comment (ashleynolan) [reply packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts]: Done!

File: packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+    it.each([
+        ['xs', '320px'],
```
Comment (xander-marjoram) [inline packages/tools/pie-css/test/scss/unit/settings/mq.spec.ts:66]: I wonder if it's possible (or desirable) to have these values somehow read from the same place as the actual definitions. If a new breakpoint were added, someone would have to remember to come and add a new test.

File: packages/tools/pie-css/scss/settings/_breakpoints.scss [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+// Breakpoints to be used by @include-media Sass library
+// https://eduardoboucas.github.io/include-media/documentation/#variable-breakpoints
+$breakpoints: (
+    xs: 320px,
+    sm: 600px,
+    md: 768px,
+    lg: 1024px,
```
Comment (xander-marjoram) [inline packages/tools/pie-css/scss/settings/_breakpoints.scss:16]: I would prefer `s`, `m`, `l`, but not sure if there's a reason why not to

File: packages/tools/pie-css/README.md [hunk 10]
```diff
@@ -82,7 +89,7 @@ createApp(App).mount('#app');
 
 ### NuxtJS
 
-If you are using NuxtJS, you can import PIE CSS in your `nuxt.config.js` file.
+If you are using NuxtJS, you can import `pie-css` in your `nuxt.config.js` file.
```
Comment (jamieomaguire) [inline packages/tools/pie-css/README.md:92]: It's actually just `Nuxt`, I think :P

File: packages/tools/pie-css/README.md [hunk 11]
```diff
@@ -121,22 +128,43 @@ If you are using Sass, you could import the CSS file as part of your styles dire
...
-### z-index
+We set `box-sizing: border-box` so that the CSS box model acts consistently when using PIE components.
 
-Some PIE Web Components use a z-index value to control how they stack on a webpage. These values are defined in the `pie-css` package as css variables and are utilized internally. In most cases, a webpage should follow the DOM's natural stacking order and the default z-index order assigned for each component. However, if you're creating custom components, refer to the following table to make sure they don't conflict with other components.
+### Typography
 
-| Token              | Z-Index Value  |
+There are a number of typographic CSS rules that we specify to help with font rendering. These are:
+
+```scss
```
Comment (jamieomaguire) [inline packages/tools/pie-css/README.md:147]:
```suggestion
```css
```

File: packages/tools/pie-css/README.md [hunk 12]
```diff
@@ -148,6 +176,133 @@ Some PIE Web Components use a z-index value to control how they stack on a webpa
 | --dt-z-index-cookie-banner | 5000           |
 | --dt-z-index-toast         | 6000           |
```
Comment (jamieomaguire) [inline packages/tools/pie-css/README.md:177]: just thinking - we might want to rename this or re-think in the future as top layer usage kind of negates any z-index here

File: packages/tools/pie-css/scss/settings/index.scss (hunk 2 above)
Comment (ashleynolan) [reply packages/tools/pie-css/scss/settings/index.scss:1]: Have added documentation of `include-media` into the `pie-css` README now

File: packages/tools/pie-css/README.md (hunk 12 above)
Comment (ashleynolan) [reply packages/tools/pie-css/README.md:182]: I won't tackle this as part of this PR (as out of scope to amend this). Let's chat about when you're in next if it's worth looking at.

File: packages/tools/pie-css/scss/settings/_breakpoints.scss (hunk 9 above)
Comment (ashleynolan) [reply packages/tools/pie-css/scss/settings/_breakpoints.scss:16]: After our conversation last week with @thejfreitas, I'll keep it as it is.

[Bootstrap](https://getbootstrap.com/docs/5.0/layout/breakpoints/#available-breakpoints) and [Material](https://mui.com/material-ui/customization/breakpoints/) both use similar breakpoint keys, so may feel more familiar to consumers.

**Review by kevinrodrigues — APPROVED** [review]

File: packages/tools/pie-css/scss/settings/_include-media.scss [hunk 13]
```diff
@@ -0,0 +1,596 @@
+//
```
Comment (jamieomaguire) [inline packages/tools/pie-css/scss/settings/_include-media.scss:1]: How do we plan on keeping this up to date?

**Review by jamieomaguire — APPROVED** [review]
I'm approving but I'm not very keen/comfortable with copying the include media package into this repo 🤔 I don't have a better suggestion at this time, but we should be mindful about how we plan to keep this updated (if it even gets updated)

**Comment by ashleynolan** [comment]
@jamieomaguire Yeh, I agree, it's not great, but it is one of the recommended ways of using the package (due to it being a mixin, so the likelihood of it changing are only in instances when fundamental Sass syntax changes, like when @use was brought in). I've just checked the NPM version history, and the previous change was 3 years before the last one (which was because of the `@use` syntax, and before that, 8 years ago, so hopefully there shouldn't be much change going forward in the package.

File: packages/tools/pie-css/scss/settings/_include-media.scss (hunk 13 above)
Comment (ashleynolan) [reply packages/tools/pie-css/scss/settings/_include-media.scss:1]: I posted some more details below – [see here](https://github.com/justeattakeaway/pie/pull/1480#issuecomment-2180192306). I think we should be pretty safe in terms of updates (I wouldn't actually expect any unless Sass updated, which we'd know about)

**Pull request merged**